pkgs/servers/nextcloud/default.nix

```diff
@@ -28,6 +28,26 @@ let
       inherit packages;
     };
 
+    outputs = [ "out" "compressed" ];
+
+    nativeBuildInputs = [ zopfli brotli ];
+
+    buildPhase = ''
+      # Create missing static gzip and brotli files
+      shopt -s globstar
+      for f in apps/** core/** dist/** resources/** themes/**; do
+        if [ -f "$f" ]; then
+          case "$f" in
+          *.css | *.js | *.json | *.svg | *.ico | *.txt | *.md | *.xml | *.html | *.ttf | *.otf | *.eot)
+            mkdir -p "$compressed/''${f%/*}"
+            zopfli --gzip -c "$f" > "$compressed/$f.gz"
+            brotli --best --keep --output="$compressed/$f.br" "$f"
```

Comment:
both command support putting the file next to the existing file. please see https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/version-management/gitea/default.nix#L76-L79